k8s-tester/secureCN/tester.go

```diff
@@ -0,0 +1,257 @@
+	"github.com/aws/aws-k8s-tester/client"
+	k8s_tester "github.com/aws/aws-k8s-tester/k8s-tester/tester"
+)
+
+type Config struct {
+	Enable bool `json:"enable"`
+
+	Stopc       chan struct{} `json:"-"`
+	Logger      *zap.Logger   `json:"-"`
+	LogWriter   io.Writer     `json:"-"`
+	Client      client.Client `json:"-"`
+	AccessKey   string        `json:"-"`
+	SecretKey   string        `json:"-"`
+	URL         string        `json:"-"`
+	ClusterName string        `json:"-"`
+}
+
+const (
+	SecureCNBundleYaml         = "securecn_bundle.yml"
+	SecureCNInstallBundle      = "install_bundle.sh"
+	SecureCNBundleTarGz        = "securecn_bundle.tar.gz"
+	SecureCNIstioAGBundleName  = "istio_init_bundle_ag.yml"
+	SecureCNAGBundleName       = "securecn_bundle_ag.yml"
+	SecureCNDoneIstioSeparator = "####### DONE ISTIO INIT PART  ########"
+	SecureCNUninstallScript    = "uninstall.sh"
+)
+
+func NewDefault() *Config {
+	return &Config{
+		Enable: false,
+	}
+}
+
+func New(cfg *Config) k8s_tester.Tester {
+	return &tester{
+		cfg: cfg,
+	}
+}
+
+type tester struct {
+	cfg *Config
+}
+
+var pkgName = path.Base(reflect.TypeOf(tester{}).PkgPath())
+
+func Env() string {
+	return "ADD_ON_" + strings.ToUpper(strings.Replace(pkgName, "-", "_", -1))
+}
+
+func (ts *tester) Name() string { return pkgName }
+
+func (ts *tester) Enabled() bool { return ts.cfg.Enable }
+
+func (ts *tester) Apply() error {
+	trueBool := true
+	falseBool := false
+	orchType := "EKS"
+
+	defer cleanInstallationFiles()
+	// create cluster
+	cmh := api_client.NewHttpClient(ts.cfg.AccessKey, ts.cfg.SecretKey, ts.cfg.URL)
+	res, err := cmh.EscherClient.CreateKubernetesCluster(context.TODO(), cmh.HttpClient, &model.KubernetesCluster{
+		AgentFailClose:                    &falseBool,
+		APIIntelligenceDAST:               &falseBool,
+		AutoLabelEnabled:                  &trueBool,
+		CiImageValidation:                 &trueBool,
+		ClusterPodDefinitionSource:        model.ClusterPodDefinitionSourceKUBERNETES,
+		EnableConnectionsControl:          &trueBool,
+		EnableVenafiIntegration:           &falseBool,
+		IsHoldApplicationUntilProxyStarts: &falseBool,
+		IsIstioIngressEnabled:             &falseBool,
+		IsMultiCluster:                    &falseBool,
+		IsPersistent:                      &falseBool,
+		Name:                              &ts.cfg.ClusterName,
+		OrchestrationType:                 &orchType,
+		PreserveOriginalSourceIP:          &falseBool,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create kubernetes cluster. %v", err)
+	}
+
+	if err := downloadAndExtractBundle(cmh, res.Payload.ID); err != nil {
+		return fmt.Errorf("failed to download and extract bundle. %v", err)
+	}
+
+	bundle, err := os.ReadFile(SecureCNBundleYaml)
+	if err != nil {
+		return fmt.Errorf("failed to read file %v. %v", SecureCNBundleYaml, err)
+	}
+	// split bundle
+	bundleStr := string(bundle)
+	bundleParts := strings.Split(bundleStr, SecureCNDoneIstioSeparator)
+	// install istio
```

Comment:
`install Istio CRDs`
All Istio components are after the `SecureCNDoneIstioSeparator`